contracts/RegisterDomains.sol

```diff
@@ -0,0 +1,55 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+
+struct DomainDetails {
+    address controller;
+    uint deposit;
+}
+
+uint256 constant DEPOSIT_PRICE = 1 ether;
+
+contract RegisterDomains {
+    address payable public owner;
+    mapping(string => DomainDetails) domains;
+
+    constructor() {
+        owner = payable(msg.sender);
+    }
+
+    function getDomain(string memory domainName) external view returns (DomainDetails memory) {
+        return domains[domainName];
+    }
+
+    function createDomain(string memory domainName) internal {
+        DomainDetails memory domain = DomainDetails({
+            controller: owner,
```

Comment:
Any sense to save the same value and spend gas for it? -3

Reply:
now it stores contract caller, so it makes sense now